ipam/ring/ring.go

```diff
@@ -524,7 +562,20 @@ func (r *Ring) PickPeerForTransfer(isValid func(mesh.PeerName) bool) mesh.PeerNa
 // Transfer will mark all entries associated with 'from' peer as owned by 'to' peer
 // and return ranges indicating the new space we picked up
 func (r *Ring) Transfer(from, to mesh.PeerName) []address.Range {
+	return r.transfer(from, to, false)
+}
+
+// Does Transfer from a dead peer
+func (r *Ring) AdminTransfer(from, to mesh.PeerName) []address.Range {
+	return r.transfer(from, to, true)
+}
+
+func (r *Ring) transfer(from, to mesh.PeerName, isAdmin bool) []address.Range {
 	r.assertInvariants()
+	defer r.trackUpdates()()
+	if isAdmin {
+		defer r.trackUpdatesOfPeer(from)()
```

Comment:
Let me see if I follow this:

- In the ordinary mode the "giving up" peer removes its rule and the "taking over" peer adds a new rule.
- In admin mode the "taking over" peer does both actions.

So, you don't really need two different versions, you can just use the second one in both cases.

It seems attractive to say "most of the time, I only modify my own rules", but I cannot see any particular reason to prefer that.

Doing it all on the "taking over" side makes guaranteeing the ordering simpler, perhaps.